### PR TITLE
[all] Split Tag & TagSet into separated files and fix missing includes.

### PR DESCRIPTION
**Pull request by damienmarchal**
______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by fredroy** [comment]
[ci-build][with-all-tests]

File: Sofa/framework/Core/src/sofa/core/objectmodel/Tag.h [hunk 1]
```diff
@@ -71,25 +64,10 @@ class SOFA_CORE_API Tag
...
-    TagSet(const Tag& t) { this->insert(t); }
-    /// Returns true if this TagSet contains specified tag
-    bool includes(const Tag& t) const { return this->count(t) > 0; }
-    /// Returns true if this TagSet contains all specified tags
-    bool includes(const TagSet& t) const;
-};
-
-} // namespace objectmodel
-
 } // namespace core
```
Comment (alxbilger) [inline Sofa/framework/Core/src/sofa/core/objectmodel/Tag.h]: Comment does not match the new namespace concatenation

File: Sofa/framework/Core/src/sofa/core/objectmodel/Tag.h [hunk 2]
```diff
@@ -98,14 +76,4 @@ struct DataTypeInfo< sofa::core::objectmodel::Tag > : public TextTypeInfo<sofa::
...
 
-template<>
-struct DataTypeInfo< sofa::core::objectmodel::TagSet > : public SetTypeInfo<sofa::core::objectmodel::TagSet >
-{
-    static const char* name() { return "TagSet"; }
-};
-
-} // namespace defaulttype
-
 } // namespace sofa
```
Comment (alxbilger) [inline Sofa/framework/Core/src/sofa/core/objectmodel/Tag.h]: Comment does not match the new namespace concatenation

File: Sofa/framework/Core/src/sofa/core/objectmodel/Tag.cpp [hunk 3]
```diff
@@ -104,9 +61,4 @@ std::istream& operator>>(std::istream& i, Tag& t)
     return i;
 }
 
-
-} // namespace objectmodel
-
-} // namespace core
-
-} // namespace sofa
+}
```
Comment (alxbilger) [inline Sofa/framework/Core/src/sofa/core/objectmodel/Tag.cpp]: Usually in SOFA the namespace is repeated in a comment after the closing curly bracket.

Comment (alxbilger) [reply Sofa/framework/Core/src/sofa/core/objectmodel/Tag.cpp]: Also in TagSet files

**Review by alxbilger — CHANGES_REQUESTED** [review]

File: Sofa/framework/Core/src/sofa/core/objectmodel/Tag.cpp (hunk 3 above)
Comment (damienmarchal) [reply Sofa/framework/Core/src/sofa/core/objectmodel/Tag.cpp]: Open question: repeating the namespace's name with comment in the ending curlybrace was a trick to improve readability dating back from when it the single line syntax (```namespace a::b::c```) was not existing and there was no super clever code editor as we have today. I'm not sure it is so much needed nowaydays.

Comment (alxbilger) [reply Sofa/framework/Core/src/sofa/core/objectmodel/Tag.cpp]: I totally agree with you and I want to campaign for removing this rule. Meanwhile, you still have discrepancies between the declaration of the namespace and the ending curly bracket comment (in TagSet for example). Either remove the comments, or fix the comments.

Comment (damienmarchal) [reply Sofa/framework/Core/src/sofa/core/objectmodel/Tag.cpp]: Sure.

**Pull request merged**